india_compliance/gst_india/doctype/purchase_reconciliation_tool/exporter.py

```diff
@@ -0,0 +1,409 @@
+                    column_widths=column_widths,
+                    bg_color=bg_color,
+                    wrap_text=wrap_text,
+                    horizontal=horizontal,
+                    vertical=vertical,
+                )
+        return ws
+
+    def update_data(self, ws, data_list, header, alignment="left", color=None):
+        for i, data in enumerate(data_list):
+            for j, value in enumerate(data):
+                column_label = ws.cell(row=2, column=j + 1).value
+                column = str(chr(64 + (j + 1)))
+                bg_color = color or "FFFFFF"
+                width = 20
+                text_color = "000000"
+
+                if "Diff" in column_label:
+                    bg_color = "f2dcdb"
+                    width = 12 if ws.title == "Invoice Data" else 18
+                    text_color = "ff0000"
+                elif j in range(7, 16):
+                    bg_color = "dce6f2"
+                    width = 12
+                elif j in range(16, 25):
+                    bg_color = "ebf1de"
+                    width = 12
+                bg_fill = self.get_pattern_fill(bg_color)
+                font = self.get_font_style(color=text_color, bold=False)
+
+                ws.cell(row=i + 3, column=j + 1).fill = bg_fill
+                ws.cell(row=i + 3, column=j + 1).value = value
+                ws.cell(row=i + 3, column=j + 1).font = font
+                ws.cell(row=i + 3, column=j + 1).number_format = "0.00"
+                ws.column_dimensions[column].width = width
+                ws.row_dimensions[1].height = 30
+                ws.row_dimensions[2].height = 30
+        return ws
+
+    def save_workbook(self, wb, file_name):
+        wb.save(file_name)
+        return wb
+
+    def read_workbook(self, file_name):
+        file_path = frappe.get_site_path("private", "files", file_name)
+        wb = openpyxl.load_workbook(file_path, read_only=False, keep_vba=True)
+        return wb
+
+    def add_worksheet(
+        self,
+        data,
+        sheet_name,
+        position,
+        sub_header_color=None,
+        body_color="FFFFFF",
+        font_size=9,
+        row=1,
+        start_column=1,
+        increment_by=1,
+        save=False,
+        workbook=None,
+        file_name=None,
+    ):
+        if not workbook:
+            workbook = self.create_workbook()
+
+        worksheet = self.create_worksheet(workbook, sheet_name, position)
+        self.append_data(
+            worksheet, data, add_row=row, add_column=start_column, is_common_header=True
+        )
+
+        # self.update_header_data(worksheet, header, row=row)
+        # self.setup_header_for_invoice_data(worksheet, range=range, color=header_color)
+        # self.update_data(worksheet, data, header, color=body_color)
+
+        if save:
+            self.save_workbook(workbook, file_name)
+
+        return workbook
+
+
+@frappe.whitelist()
+def export_data_to_xlsx(args, column_widths=None):
```

Comment:
Organising modules plays an important part(very similar to what should a function do and not do).

exporter.py is a helper module for reco_tool, but parent whitelisted function should be always in core module where you can have a complete picture.